src/functionality.js

```diff
@@ -0,0 +1,144 @@
+    check.setAttribute('id', `${task.index}`);
+
+    const description = document.createElement('p');
+    description.textContent = task.description;
+    description.className = 'description';
+    description.setAttribute('contenteditable', 'true');
+
+    const trash = document.createElement('i');
+    trash.className = 'fa-solid fa-trash-can fa-beat';
+
+    const anchor = document.createElement('i');
+    anchor.className = 'fa-solid fa-ellipsis-vertical';
+
+    topLeft.append(check, description, trash);
+    topRight.appendChild(anchor);
+    leftRight.append(topLeft, topRight);
+
+    const ruler = document.createElement('hr');
+
+    taskPane.append(leftRight, ruler);
+
+    domTasks.appendChild(taskPane);
+
+    store();
+
+    topLeft.addEventListener('mouseenter', () => {
+      trash.style.display = 'block';
+      anchor.style.display = 'none';
+    });
+
+    topLeft.addEventListener('mouseleave', () => {
+      trash.style.display = 'none';
+      anchor.style.display = 'block';
+    });
+
+    description.addEventListener('input', () => {
+      const newValue = description.textContent;
+      const test = tasks.some((tester) => {
+        if (tester.description.toLowerCase() === newValue.toLowerCase()) return true;
+        return false;
+      });
+      if (test) {
+        abnormal.style.display = 'block';
+        abnormal.innerHTML = 'Sorry! You can\'t add the same task twice';
+      } else {
+        abnormal.style.display = 'none';
+        task.description = newValue;
+        store();
+      }
+    });
+
+    trash.addEventListener('click', () => {
+      if (tasks.length === 1) {
+        tasks = [];
+        store();
+        domTasks.innerHTML = '';
+      } else {
+        const indices = i;
+        tasks = tasks.filter((elem) => elem.index - 1 !== indices);
+        tasks = tasks.map((mapped, i) => ({
+          description: `${mapped.description}`,
+          index: `${i + 1}`,
+          completed: false,
+        }));
+        domTasks.innerHTML = '';
+        displayTasks();
+      }
+    });
+  });
+};
+
+const getter = () => {
+  tasks = JSON.parse(localStorage.getItem('tasks'));
+  if (tasks) {
+    displayTasks();
+  } else {
+    tasks = [];
+  }
+};
+
+const addTask = () => {
+  if (todo.value) {
+    const test = tasks.some((tester) => {
+      if (tester.description.toLowerCase() === todo.value.toLowerCase()) return true;
+      return false;
+    });
+    if (test) {
+      abnormal.style.display = 'block';
+      abnormal.innerHTML = 'Sorry! You can\'t add the same task twice';
+    } else {
+      abnormal.style.display = 'none';
+      const task = {
+        description: todo.value,
+        index: tasks.length + 1,
+        completed: false,
+      };
+      tasks.push(task);
+      todo.value = '';
+      domTasks.innerHTML = '';
+      displayTasks();
+    }
+  } else {
+    abnormal.style.display = 'block';
+    abnormal.innerHTML = 'Sorry! You can\'t add an empty task';
+  }
+};
```

Comment:
- [ ] @Bestbynature great work trying to work on the changes requested, however, I see you only changed the file name, kindly consider renaming the file index.js and then move the add and delete functionality to a new file which will be imported to the index.js file as previously requested.